### PR TITLE
Added Restaurant Backend

### PR DESCRIPTION
**Pull request by masalemi**
This PR contains all of the information for the Restaurant Backend. It includes all logic needed to store and request restaurants and their data.

Things that still need to be done to integrate into front end:
1. Replace add-restaurant.html with a way to create restaurants using an ID based off of google maps.
2. Integrate the backend data printed at the bottom of restaurant-page.html into the page nicely.

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/RestaurantServlet.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+  private Datastore datastore;
+
+  @Override
+  public void init() {
+    datastore = new Datastore();
+  }
+
+  /**
+   * Responds with the "about me" section for a particular user.
```
Comment (zrolston) [inline src/main/java/com/google/codeu/servlets/RestaurantServlet.java]: Is this comment still valid?

Comment (masalemi) [reply src/main/java/com/google/codeu/servlets/RestaurantServlet.java]: Fixed!

**Review by jenniferwang99 — APPROVED** [review]
Thanks for doing this!

**Comment by masalemi** [comment]
Good point! I noticed this after the PR was merged so I fixed it and it is included in my current open PR.

**Pull request merged**